scipy/spatial/transform/_rotation.pyx

```diff
@@ -1107,13 +1117,35 @@ cdef class Rotation:
         # If a single matrix is given, convert it to 3D 1 x 3 x 3 matrix but
         # set self._single to True so that we can return appropriate objects in
         # the `to_...` methods
-        cdef double[:, :, :] cmatrix
         if matrix.shape == (3, 3):
-            cmatrix = matrix[None, :, :]
+            matrix = matrix[np.newaxis, :, :]
             is_single = True
-        else:
-            cmatrix = matrix
 
+        # Calculate the determinant of the rotation matrix
+        # (should be positive for right-handed rotations)
+        dets = np.linalg.det(matrix)
+        if np.any(dets <= 0):
+            ind = np.where(dets <= 0)[0][0]
+            raise ValueError("Non-positive determinant (left-handed or null "
+                             f"coordinate frame) in rotation matrix {ind}: "
+                             f"{matrix[ind]}.")
+
+        # Gramian orthonormality check
+        # (should be the identity matrix for orthonormal matrices)
+        grams = matrix @ np.swapaxes(matrix, 1, 2)
```

Comment:
`grams` to something more readable? Or simply move into `np.isclose`.

Reply:
`gramian` ok? https://en.wikipedia.org/wiki/Gram_matrix